src/lowercase.rs

```diff
@@ -16,10 +23,10 @@ impl<'a> Lowercase<'a> {
     /// Take a string and return the lowercased version.
     pub fn new(s: &'a str) -> Self {
         // Avoid allocating if it's not necessary
-        if s.chars().all(char::is_lowercase) {
-            Lowercase(Cow::Borrowed(s))
+        if s.chars().any(|c| c.is_ascii_uppercase()) {
```

Comment:
does char::is_ascii_uppercase work here too?

Reply:
No it complained about a missing `for<'a>` and I didn't understand its problem

Reply:
yeah i've encountered that too... surprising since it basically worked in `all(char::is_ascii_lowercase)`